web/app/js/components/Navigation.jsx

```diff
@@ -0,0 +1,329 @@
+    width: theme.spacing.unit * 7,
+    [theme.breakpoints.up('sm')]: {
+      width: theme.spacing.unit * 9,
+    },
+  },
+  toolbar: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'flex-end',
+    padding: '0 8px',
+    ...theme.mixins.toolbar,
+  },
+  content: {
+    flexGrow: 1,
+    width: `calc(100% - ${drawerWidth}px)`,
+    backgroundColor: theme.palette.background.default,
+    padding: theme.spacing.unit * 3,
+  },
+  linkerdLogoContainer: {
+    backgroundColor: theme.palette.primary.dark,
+  },
+  linkerdNavLogo: {
+    minWidth: "180px",
+  },
+  navMenuItem: {
+    paddingLeft: "24px",
+    paddingRight: "24px",
+  },
+});
+
+class NavigationBase extends React.Component {
+  constructor(props) {
+    super(props);
+    this.api = this.props.api;
+    this.handleApiError = this.handleApiError.bind(this);
+
+    this.state = this.getInitialState();
+  }
+
+  getInitialState() {
+    return {
+      drawerOpen: true,
+      resourceMenuOpen: false,
+      helpMenuOpen: false,
+      latestVersion: '',
+      isLatest: true,
+      namespaceFilter: "all"
+    };
+  }
+
+  componentDidMount() {
+    this.fetchVersion();
+  }
+
+  fetchVersion() {
+    let versionUrl = `https://versioncheck.linkerd.io/version.json?version=${this.props.releaseVersion}&uuid=${this.props.uuid}&source=web`;
+    this.versionPromise = fetch(versionUrl, { credentials: 'include' })
+      .then(rsp => rsp.json())
+      .then(versionRsp => {
+        let latestVersion;
+        let parts = this.props.releaseVersion.split("-", 2);
+        if (parts.length === 2) {
+          latestVersion = versionRsp[parts[0]];
+        }
+        this.setState({
+          latestVersion,
+          isLatest: latestVersion === this.props.releaseVersion
+        });
+      }).catch(this.handleApiError);
+  }
+
+  handleApiError(e) {
+    this.setState({
+      error: e
+    });
+  }
+
+  handleDrawerOpen = () => {
+    this.setState({ drawerOpen: true });
+  };
+
+  handleDrawerClose = () => {
+    this.setState({ drawerOpen: false });
+  };
+
+  handleResourceMenuClick = () => {
+    this.setState(state => ({ resourceMenuOpen: !state.resourceMenuOpen }));
+  };
+
+  handleHelpMenuClick = () => {
+    this.setState(state => ({ helpMenuOpen: !state.helpMenuOpen }));
+  }
+
+  menuItem(path, title, icon) {
+    const { classes, api } = this.props;
+    let normalizedPath = this.props.location.pathname.replace(this.props.pathPrefix, "");
+    let isCurrentPage = path => path === normalizedPath;
+
+    return (
+      <MenuItem
+        component={Link}
+        to={api.prefixLink(path)}
+        className={classes.navMenuItem}
+        selected={isCurrentPage(path)}>
+        <ListItemIcon>{icon}</ListItemIcon>
+        <ListItemText primary={title} />
+      </MenuItem>
+    );
+  }
+  render() {
+    const { classes, ChildComponent } = this.props;
+
+    return (
+      <div className={classes.root}>
+        <AppBar
+          position="absolute"
```

Comment:
Could this line be moved to the `appBar` style object?

Reply:
This is part of `AppBar`'s API:
https://material-ui.com/api/app-bar/#appbar